modules/scrollback/index.js

```diff
@@ -110,108 +111,389 @@ module.exports.formatLine = function(line) {
   }
 };
 
+/* Abstract: Quoting lines from IRC is an essential problem with far-reaching
+ * impact: many humans and semi-sentient AIs depend on their ability to precisely
+ * and concisely specify exactly which embarrasing statements and/or nonsequitirs
+ * presented in a real-time group chat system are worthy of being selected for
+ * processing by other software, including various quotation recording libraries,
+ * machine learning algorithms, and text processing utilities. In this function,
+ * the authors present a novel approach to the formatted text selection problem:
+ * a deterministic specification of a flexible language for selecting text is
+ * defined based on three criteria: speaker name, regular expression match, and
+ * sequential number (indices in an inverse relationship with current time); then
+ * an algorithm for parsing the language specification and selecting the lines a
+ * user's instance of the specification matches is demonstrated.
+ */
+/* Get lines from scrollback using various range specifications
+ *
+ * There are three possible range specifiers: nick (N), regex (R), and line number (L).
+ * The formal grammar for these specifications:
+ *   All = Spec | All Spec | All, Spec
+ *   Spec = Nickopt Regexopt Lineopt
+ *   Nickopt = e | Nicks
+ *   Regexopt = e | Regexes
+ *   Lineopt = e | Lines
+ *   Nicks = N | Nicks,N
+ *   Regexes = R | Regexes,R
+ *   Lines = L | L-L | L..L | L...L | Lines,Lines
+ *
+ * Line parsing is handled by https://github.com/euank/node-parse-numeric-range
+ *
+ * In general, a range specification after being parsed by numeric-range looks like:
+ *   [N1[,N2][,N3]...[,Ni]] [R1[,R2][,R3]...[,Rj]] [L1[,L2][,L3]...[,Lk]][,]
+ * This parses to a cross-product of ranges:
+ *   [N1,R1,L1], [N1,R1,L2], ..., [N1,R1,Lk],
+ *   [N1,R2,L1], [N1,R2,L2], ..., [N1,R2,Lk],
+ *   ...
+ *   [N1,Rj,L1], ..
+ *   (and so on and so forth for N2,...,Ni).
+ *
+ * If no nick is specified, lines from every nick will be considered;
+ * if no regex is specified, every line will match;
+ * if no line is specified, the first line matching the range is returned.
+ *
+ * Multiple range specs may be concatenated together.
+ * In cases where ambiguity could occur, a trailing , can be used to indicate
+ * where one range ends and the next begins.
+ *
+ * Examples:
+ * joe /dog/ 3
+ * -> Return the third line fro user joe matching /dog/
+ *
+ * joe /dog/,/cat/
```

Comment:
Mind adding an example? I *think* that `joe /dog/ /cat/` matches any line for cat, not just joe lines, but I'd like the correct answer as an example.